sources/packages/backend/apps/api/src/route-controllers/application/application.controller.service.ts

```diff
@@ -822,27 +823,62 @@ export class ApplicationControllerService {
     applicationId: number,
     options?: { studentId?: number },
   ): Promise<ApplicationOverallDetailsAPIOutDTO> {
-    try {
-      const applications =
-        await this.applicationService.getAllApplicationVersions(applicationId, {
-          studentId: options?.studentId,
-        });
-      return {
-        previousVersions: applications.map((application) => ({
-          id: application.id,
-          submittedDate: application.submittedDate,
-          applicationEditStatus: application.applicationEditStatus,
-        })),
-      };
-    } catch (error: unknown) {
-      if (
-        error instanceof CustomNamedError &&
-        error.name === APPLICATION_NOT_FOUND
-      ) {
-        throw new NotFoundException("Application not found.");
-      }
-      throw error;
+    const application = await this.applicationService.getAllApplicationVersions(
+      applicationId,
+      {
+        studentId: options?.studentId,
+      },
+    );
+    if (!application) {
+      throw new NotFoundException("Application not found.");
+    }
+    // Current application, the only one that will not have its main status as 'Edited'.
+    const currentApplication =
+      this.transformToApplicationOverallDetailsAPIOutDTO(application);
+    // Convert all the past versions of the application.
+    let previousVersions = application.parentApplication.versions.map(
+      (version) => this.transformToApplicationOverallDetailsAPIOutDTO(version),
+    );
+    // Check if there is an in-progress change request for the application.
+    const inProgressChangeRequest = previousVersions.find((version) =>
+      APPLICATION_EDIT_STATUS_IN_PROGRESS_VALUES.includes(
+        version.applicationEditStatus,
+      ),
+    );
+    // Remove the in-progress change request from the previous versions list.
+    // A maximum of one in-progress change request is allowed.
+    if (inProgressChangeRequest) {
+      previousVersions = previousVersions.filter(
+        (version) => version.id !== inProgressChangeRequest.id,
+      );
```

Comment:
We can do an early return if the previous versions are not present.

Reply:
The goal of the early return is to reduce nesting, complexity, and make the code easy to read.
The nesting would not be reduced in this situation. I added the early return to see how it looks and it does not seems like an improvement in this scenario. Please let me know if I am missing something at a level that it would be a blocker.